app/models/theme.rb

```diff
@@ -125,16 +121,23 @@ def self.clear_default!
   end
 
   def self.transform_ids(ids, extend: true)
-    return [] if ids.blank?
+    get_set_cache "transformed_ids_#{ids.join("_")}" do
+      next [] if ids.blank?
+
+      ids.uniq!
+      parent = ids.first
```

Comment:
One thing to consider here is using `Array#shift`. It will return the first value and mutate the array to remove it which is I think cleaner than the way you're doing it.